python/lzreposync/src/lzreposync/db_utils.py

```diff
@@ -0,0 +1,37 @@
+#  pylint: disable=missing-module-docstring
+
+from lzreposync.repo_dto import RepoDTO
+from spacewalk.server import rhnSQL
+
+
+def get_repositories_by_channel_id(channel_id):
+    """
+    Fetch repositories information form the database, and return a list of
+    RepoDTO objects
+    """
+    rhnSQL.initDB()
+    h = rhnSQL.prepare(
+        """
+        select s.id, s.source_url, s.metadata_signed, s.label as repo_label, cst.label as repo_type_label
+        from rhnContentSource s,
+             rhnChannelContentSource cs,
+             rhnContentSourceType cst
+        where s.id = cs.source_id
+          and cst.id = s.type_id
+          and cs.channel_id = :channel_id"""
```

Comment:
You could join with the `rhnChannel` table and get the repositories by channel label. In which case you would need to copy the channel architecture in each `RepoDTO`